frame/aura/src/lib.rs

```diff
@@ -219,10 +220,14 @@ impl<T: Config> OneSessionHandler<T::AccountId> for Pallet<T> {
 			let next_authorities = validators.map(|(_, k)| k).collect::<Vec<_>>();
 			let last_authorities = Self::authorities();
 			if last_authorities != next_authorities {
-				let bounded = <WeakBoundedVec<_, T::MaxAuthorities>>::force_from(
-					next_authorities,
-					Some("AuRa new session"),
-				);
+				if next_authorities.len() as u32 > T::MaxAuthorities::get() {
```

Comment:
Isn't a better solution here one that is a bit more comprehensive? Extend the `OnSessionHandler` to have a const for `MaxAuthorities`, then we don't use an iterator, but instead expect a bounded vec in the API, and then no truncation is necessary?

I mean, this also does not seem that bad to me to just truncate, but seems to me if we would re-write this API to begin with, it would look different.

Reply:
Rewriting the `OneSessionHandler` API sounds good too, I'll need a bit more context however, but will attempt in another PR.

Reply:
I think is is very bad because it is unexpected! In other places genesis just panics, so making this fail without a panic is super confusing for the end users.

Reply:
Correct me if I'm wrong, but `on_new_session` is run whenever there's a new session, so it's not necessarily _just_ on genesis which this function is called, hence we don't panic here. The previous code also did not panic using `force_from`, but in places where the function is run _only_ on genesis (e.g. `initialize_authorities`), then we do panic.

Reply:
Correct, sorry I misread the code. Makes sense.